### ISSUE TITLE
Opencl blas dot function crashing

### ISSUE THREAD

**Issue by Kerkil** [arrayfire/arrayfire#349]
```
void testAFDot()
{
    std::cout << "BLAS dot is tested." << std::endl;
    try{
        const int nr = 4;

        float hA[] = {1,2,3,4};
        float hB[] = {2,3,4,5};

        af::array A(nr,hA);
        af::array B(nr,hB);
        
        auto C = af::dot(A, B, AF_TRANSPOSE, AF_NO_TRANSPOSE);

    } catch (af::exception &e){
        std::cout << e.what() << std::endl;
        throw;
    }
}
```
This code crashes without showing the exception:
```
Starting /Users/recondev/Projects/build-Linspace-osx64-Debug/Linspace...
BLAS dot is tested.
The program has unexpectedly finished.
/Users/recondev/Projects/build-Linspace-osx64-Debug/Linspace crashed
```
Looks like an error related to #338.

**Comment by pavanky** [comment]
@Kerkil Can you check if this error still persists or if it has been resolved by #338  ?

**Comment by Kerkil** [comment]
Yup. This was originating from the same issue and has been successfully resolved.
